specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/storageTaskAssignments.json

```diff
@@ -0,0 +1,840 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "StorageManagementClient",
+    "description": "The Azure Storage Management API.",
+    "version": "2023-05-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/storageTaskAssignments/{storageTaskAssignmentName}": {
+      "put": {
+        "tags": [
+          "StorageTaskAssignments"
+        ],
```

Comment:
Could you please fix the Swagger Lint errors flagged here  as must fix  
[https://github.com/Azure/azure-rest-api-specs/pull/28461](https://github.com/Azure/azure-rest-api-specs/pull/28461)

[must fix]The following errors/warnings are introduced by current PR:

Reply:
Please also fix the warnings

Reply:
@raosuhas I am confused by the Lint error. It mentions PUT can only have response codes 200 and 201. However, this is a long running operation, so we initially need to return 202 with location header, and then the final state is 200/201. Like described here, https://learn.microsoft.com/en-us/rest/api/fabric/articles/long-running-operation

Reply:
Regarding the warnings:
1) EnumInsteadOfBoolean - the three properties it is flagging (allowAclAuthorization, isNFSv3Enabled, enabled) are all self-explanatory by their names (and have additional descriptions provided). It does not make sense to convert these to enums when the only possible values are true and false. We have conducted all our POCs and written our policies in this manner using booleans.
2) MissingSegmentsInNestedResourceListOperation - I do not understand this warning, we have a GET call in all the APIs. "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}" has its own APIs in the storage.json file.

Reply:
The official ARM RPC documentation for LRO is here : https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/async-api-reference.md#creating-or-updating-resources-asynchronously 

For puts even for LRO operations , the 202 is not the recommended pattern

Reply:
1. If the Booleans do not make sense then you can ignore. But please consider future extensibility before ignoring these warning. Would there be other states in the future ? It is very unwieldy to author an api with a large list of boolean params. 
2. You can ignore the MissingSegmentsInNestedResourceListOperation  for storageTaskAssignments. This seems like a bug in the automation 
3. For     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/reports": 

The modelling is not clear. reports must be a resource collection so you need to define individual gets for each report as well.

Reply:
> The official ARM RPC documentation for LRO is here : https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/async-api-reference.md#creating-or-updating-resources-asynchronously
> 
> For puts even for LRO operations , the 202 is not the recommended pattern

We already have systems integrated and we are learning about this new model only now. The feature has been in Preview, and we have portal and backend all implemented with PUT returning 202.

In the doc you linked above, it is mentioned: 

> "Some of the resource provider's existing PUT APIs return 202 Accepted. Please note this is a old model. For PUT requests, 201 / 200 + ProvisioningState is the preferred model."

Could you kindly consider this as an old existing model and give us an exception for this flagged error.

Reply:
> 3. For     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/reports":
> 
> The modelling is not clear. reports must be a resource collection so you need to define individual gets for each report as well.

A little context regarding this API:
This is an async operation that Lists the generated results of multiple assignment runs/instances (at the storage account level). The customer never creates an instance of a report, instead they create an assignment (for which we have CRUD), and when they enable such assignment, that will generate some results. Which is why we only have get/list reports the customer can call to see all the results.

We are following the same pattern here: https://github.com/Azure/azure-rest-api-specs/blob/b1ee402ca88a56d01cc61a4cd9ca28f9bb5bf617/specification/storageactions/resource-manager/Microsoft.StorageActions/stable/2023-01-01/storageactions.json#L406

And this was actually a recommendation by ARM team and also approved.

Reply:
thanks that sounds fine, but my point is that "StorageTaskReportInstance" is defined as a ProxyResource , which means it will have an id associated with it. This means that customers and clients could call GET on individual reports right ? So you would need a route for that as well ?

Reply:
Hey @raosuhas - The instances generated are server facing, if you look at the example the name of the resource is server generated and follows the pattern "instanceX" ; if the list has 5 items, meaning 5 executions on server side, the response shall have names, literally as : instance1, instance2 .. instance 5 , so the customer does not really have a way to fetch any insatance of this proxy type.

In other words, customer does not own this type, they dont really have a way to know the name of the report instance.
This idea was proposed by ARM team, as pointed by Tricia above :)

Reply:
If the name is of the format, instanceX (even though it is generated from server side), the user can create a request with this resource Id with instanceX format (or understand that this is the format ) and do a GET on it.   

Could you comment on what will happen in those cases ?

Reply:
instanceX is a randomly generated name and assigned to the most recent run, it is not something customer can designate.
 Server returns most recent run first : run3, run2, run1 -> where run3 is most recent and the response will have instance1 -> run3 , instance2 -> run2, instance3 -> run1
 
 Given that if a new run happens, say run4, then instance 1 will now point to run4 
 
 Customer cannot designate the instance towards a specific run, it is read only and hence only supports LIST.
 Following the same guideline as here, where we added the same API for Microsoft.StorageActions /reports endpoint
 
 [azure-rest-api-specs/specification/storageactions/resource-manager/Microsoft.StorageActions/stable/2023-01-01/storageactions.json](https://github.com/Azure/azure-rest-api-specs/blob/b1ee402ca88a56d01cc61a4cd9ca28f9bb5bf617/specification/storageactions/resource-manager/Microsoft.StorageActions/stable/2023-01-01/storageactions.json#L406)